src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1639,10 +1639,18 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                                             } else {
                                                 self.get_default_err_msg(place)
                                             };
+                                            let sp = self.mir.source_info(locations[0]).span;
+                                            let mut help_stmt = String::new();
+                                            if let Ok(src) =
+                                                self.tcx.sess.codemap().span_to_snippet(sp) {
+                                                    help_stmt = format!(
+                                                        "consider changing this to be a \
+                                                        mutable reference: `&mut {}`", &src[1..]);
+                                            };
                                             err_info = Some((
```

Comment:
`err_info` will need to include the contents of `&src[1..]` above in order to supply to use it in the suggestion.